osquery/dispatcher/scheduler.cpp

```diff
@@ -184,15 +198,29 @@ void SchedulerRunner::start() {
     if ((i % 3) == 0) {
       relayStatusLogs(true);
     }
-
-    // Put the thread into an interruptible sleep without a config instance.
-    pauseMilli(interval_ * 1000);
+    auto loop_step_duration =
+        std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::steady_clock::now() - start_time_point);
+    if (time_drift_ > max_time_drift) {
+      // giving up
+      time_drift_ = std::chrono::milliseconds::zero();
+    }
+    if (loop_step_duration + time_drift_ < interval_) {
+      pauseMilli(interval_ - loop_step_duration - time_drift_);
+      time_drift_ = std::chrono::milliseconds::zero();
+    } else {
+      time_drift_ += loop_step_duration - interval_;
```

Comment:
It would make the code easier to read to move the max_time_drift check here.

Reply:
ok, done